lib/modules/smart_scheduling_links/test/smart_scheduling_basic_test.rb

```diff
@@ -1,47 +1,1150 @@
+      sequence = @sequence_class.new(instance_copy, @client)
+
+      assert_raises(Inferno::AssertionException) do
+        sequence.run_test(@manifest_downloadable_test)
+      end
+    end
+
+    it 'succeeds when manfest file is valid json' do
+      valid_json = '{"test" : "test"}'
+      stub_request(:get, @manifest_url)
+        .to_return(status: 200, body: valid_json, headers: {})
+
+      instance_copy = @instance.clone
+      instance_copy.manifest_url = @manifest_url
+      sequence = @sequence_class.new(instance_copy, @client)
+
+      sequence.run_test(@manifest_downloadable_test)
+    end
+  end
+
+  # 03
+  describe 'manifest is structured properly' do
+    before do
+      @manifest_minimum_requirement_test = @sequence_class[:manifest_minimum_requirement]
+    end
+
+    it 'succeeds with manifest file containing each required field' do
+      instance_copy = @instance.clone
+      sequence = @sequence_class.new(instance_copy, @client)
+      sequence.instance_variable_set(:@manifest, @sample_manifest_file)
+      assert_raises(Inferno::PassException) do
+        sequence.run_test(@manifest_minimum_requirement_test)
+      end
+    end
+
+    it 'fails when missing transactionTime' do
+      instance_copy = @instance.clone
+      sequence = @sequence_class.new(instance_copy, @client)
+
+      sample_manifest_file_missing_transaction_time = JSON.parse('{
+        "request" : "http://www.example.com/$bulk-publish",
+        "output" : [
+          {
+            "type": "Location",
+            "url": "http://www.example.com/locations.ndjson"
+          },
+          {
+            "type": "Schedule",
+            "url": "http://www.example.com/schedules.ndjson"
+          },
+          {
+            "type": "Slot",
+            "url": "http://www.example.com/slots-2021-W09.ndjson",
+            "extension": {
+              "state": [
+                "MA"
+              ]
+            }
+          }
+        ]
+      }')
+
+      sequence.instance_variable_set(:@manifest, sample_manifest_file_missing_transaction_time)
+
+      assert_raises(Inferno::AssertionException) do
+        sequence.run_test(@manifest_minimum_requirement_test)
+      end
+    end
+
+    it 'fails when missing request' do
+      instance_copy = @instance.clone
+      sequence = @sequence_class.new(instance_copy, @client)
+
+      sample_manifest_file_missing_request = JSON.parse('{
+        "transactionTime" : "2021-03-10T18:16:34.535Z",
+        "output" : [
+          {
+            "type": "Location",
+            "url": "http://www.example.com/locations.ndjson"
+          },
+          {
+            "type": "Schedule",
+            "url": "http://www.example.com/schedules.ndjson"
+          },
+          {
+            "type": "Slot",
+            "url": "http://www.example.com/slots-2021-W09.ndjson",
+            "extension": {
+              "state": [
+                "MA"
+              ]
+            }
+          }
+        ]
+      }')
```

Comment:
Something that would cut down a bit on code is put a single "good" Location, Slot and Schedule file in fixtures, load them up in the 'before', then just tweaking them as needed for each failure case.  That would turn 26 lines down into about 1.  Also, for your awareness, you can just use Ruby's literal notation instead of JSON and then parsing the JSON.  But I would recommend moving this fixture into the fixtures directory myself.  Look in another test for an example of loading a fixture.

Reply:
looking into it...

Reply:
Added fixture